e2e/compatibility_test.go

```diff
@@ -0,0 +1,102 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+
+	"gotest.tools/fs"
+	"gotest.tools/icmd"
+)
+
+func importAndTagImage(t *testing.T, cmd icmd.Cmd, tmpDir *fs.Dir, tag string, url string) string {
+
+	icmd.RunCmd(icmd.Command("wget", "-q", "-O", tmpDir.Join("img.tar"), url)).Assert(t, icmd.Success)
+
+	cmd.Command = dockerCli.Command("load", "-i", tmpDir.Join("img.tar"))
+	result := icmd.RunCmd(cmd).Assert(t, icmd.Success)
+	var digest string
+	for _, line := range strings.Split(result.Combined(), "\n") {
+		if strings.Contains(line, "sha256:") {
+			digest = strings.Split(line, "sha256:")[1]
+		}
+	}
+	digest = strings.Trim(digest, " \r\n")
+
+	cmd.Command = dockerCli.Command("tag", digest, tag)
+	icmd.RunCmd(cmd).Assert(t, icmd.Success)
+
+	return "sha256:" + digest
+}
+
+func getBundle(t *testing.T, url string) []byte {
+	client := http.Client{Timeout: time.Minute * 1}
+	res, err := client.Get(url)
+	assert.NilError(t, err)
+
+	body, err := ioutil.ReadAll(res.Body)
+	assert.NilError(t, err)
+
+	return body
+}
+func TestBackwardsCompatibilityV1(t *testing.T) {
+	runWithDindSwarmAndRegistry(t, func(info dindSwarmAndRegistryInfo) {
+
+		cmd := info.configuredCmd
+		appName := "app-e2e"
+		tmpDir := fs.NewDir(t, appName)
+		//defer tmpDir.Remove()
+
+		bundle := getBundle(t, "https://github.com/docker/app-e2e/raw/master/bundles/v0.9.0/bundle.json")
+
+		invocDigest := importAndTagImage(t, cmd, tmpDir, "app-e2e:0.1.0-invoc", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/app-e2e-invoc.tar")
+		backendDisgest := importAndTagImage(t, cmd, tmpDir, "app-e2e/backend", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/backend.tar")
+		frontendDigest := importAndTagImage(t, cmd, tmpDir, "app-e2e/frontend", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/frontend.tar")
+
+		bundle = bytes.ReplaceAll(bundle, []byte("<digest-app-e2e-invoc>"), []byte(invocDigest))
```

Comment:
I wonder why we need that. Shouldn't it be a relocation map instead? WDYT @eunomie ?

Reply:
For what I understand it's not in link with the relocation map, it's more the bundle is "constructed" here. The stored bundle is not complete. If the digest after a load is constant so we should just put it in the `bundle.json`
If not and we need to update it here, I think we can improve the comprehension just by changing the names of the variables.
Something like:

```go
bundleTemplate := getBundle(t, "...bundle.json.template")
invocDigest := importAndTagImage()
backendDigest := importAndTagImage()
frontentDigest := importAndTagImage()

bundle := createBundle(bundleTemplate, invocDigest, backendDigest, frontendDigest)
```

Something to really expose we are building it here.